### PR TITLE
feat(connector): [BOA/Cyb] Include merchant metadata in capture and void requests

### PR DESCRIPTION
**Pull request by deepanshu-iiitu**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Original PR: https://github.com/juspay/hyperswitch/pull/3308
Include merchant metadata in capture and void requests for connectors BOA and Cybersource.

Note: This PR only contains the fix for BOA because Cybersource Metadata changes are not present in this version.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

https://github.com/juspay/hyperswitch/issues/3307

## How did you test it?

Testing can be done by creating 2 manual authorized payments for Bankofamerica. While creating these payments you should pass metadata in the requests.
```
"metadata": {
    "count_tickets": "1",
    "transaction_number": 2233
 },
```
Then you should void one payment and capture the other. You should then be able to see this metadata in merchantDefinedInformation in BOA Dashboard.

Screenshot 2024-01-09 at 11 40 17 AM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by Narayanbhat166 — APPROVED** [review]

File: crates/router/src/types.rs [hunk 1]
```diff
@@ -419,6 +419,7 @@ pub struct PaymentsCaptureData {
     pub multiple_capture_data: Option<MultipleCaptureRequestData>,
     pub connector_meta: Option<serde_json::Value>,
     pub browser_info: Option<BrowserInformation>,
+    pub metadata: Option<pii::SecretSerdeValue>,
```
Comment (ArjunKarthik) [inline crates/router/src/types.rs:422]: Please add comment on from where and what info this metadata holds?

File: crates/router/src/types.rs [hunk 2]
```diff
@@ -533,6 +534,7 @@ pub struct PaymentsCancelData {
     pub cancellation_reason: Option<String>,
     pub connector_meta: Option<serde_json::Value>,
     pub browser_info: Option<BrowserInformation>,
+    pub metadata: Option<pii::SecretSerdeValue>,
```
Comment (ArjunKarthik) [inline crates/router/src/types.rs:537]: Please add comment on from where and what info this metadata holds?

File: crates/router/src/types.rs (hunk 1 above)
Comment (deepanshu-iiitu) [reply crates/router/src/types.rs:422]: Done

File: crates/router/src/types.rs (hunk 2 above)
Comment (deepanshu-iiitu) [reply crates/router/src/types.rs:538]: Done

**Review by ArjunKarthik — DISMISSED** [review]

**Review by Narayanbhat166 — DISMISSED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by vspecky — APPROVED** [review]

**Review by vspecky — APPROVED** [review]

**Pull request merged**